libqtile/window.py

```diff
@@ -1316,3 +1316,12 @@ def cmd_set_position(self, dx, dy, curx, cury):
                 self.group.layout.focused = index2
                 self.group.layoutAll()
                 break
+
+    def __getstate__(self):
```

Comment:
I don't directly pickle a `window` object. However as there are many attributes of `layout`  class which are `window` class objects.When I pickle a `layout` class , they get pickled as well. 

The `object` class has many links to `group` class and `layout` class. So it is not possible to pickle the window state in its original form. Using the fact that after restart the `qtile` object has a `windowMap` attribute which I can use to restore the window state back to original by just using the wid .Hence I used this set of  `__gestate__` and `__setstate__`  functions.